src/Core/FeatureFlag/Handler/HandlerFactory.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\FeatureFlag\Handler;
+
+use PrestaShop\PrestaShop\Core\FeatureFlag\FeatureFlagSettings;
+use PrestaShop\PrestaShop\Core\FeatureFlag\TypeHandlerInterface;
+use PrestaShopBundle\Entity\FeatureFlag;
+
+class HandlerFactory
+{
+    /**
+     * Get next handler that can be used with this feature flag.
+     *
+     * @param FeatureFlag $featureFlag
+     *
+     * @return TypeHandlerInterface
+     *
+     * @throws \PrestaShopException
+     */
+    public static function getHandler(FeatureFlag $featureFlag): TypeHandlerInterface
+    {
+        foreach ($featureFlag->getTypeOrder() as $type) {
+            switch ($type) {
+                default:
+                case FeatureFlagSettings::TYPE_DB:
+                    $handler = new DbHandler($featureFlag);
+                    break;
+                case FeatureFlagSettings::TYPE_ENV:
+                    $handler = new EnvHandler($featureFlag);
+                    break;
+                case FeatureFlagSettings::TYPE_DOTENV:
+                    $handler = new DotEnvHandler($featureFlag);
```

Comment:
Handlers are services. They should be injected in the constructor instead of being instantiated in this class.
You can use tag system to inject every handlers all at once.

If you need more explanation, feel free to ping me :)